### PR TITLE
write a github workflow summary of the publishing status

### PR DESCRIPTION
**Pull request by devoncarew**
- write a github workflow summary of the publishing status
- allow creating PR comments to fail w/o failing the build
- fix https://github.com/dart-lang/ecosystem/issues/56

### CODE REVIEW

File: pkgs/firehose/lib/src/github.dart [hunk 1]
```diff
@@ -43,6 +43,20 @@ class Github {
 
   http.Client get httpClient => _httpClient ??= http.Client();
 
+  /// Write the given [markdownSummary] content to the GitHub
+  /// `GITHUB_STEP_SUMMARY` file. This will cause the markdown output to be
```
Comment (kevmoo) [inline pkgs/firehose/lib/src/github.dart:47]: not a HUGE issue, but if these features are documented somewhere, it'd be cool to include reference in the source – just so folks understand...

**Comment by kevmoo** [comment]
How is this summary different than the summary that shows up on the PR page? Duplicate?

**Comment by devoncarew** [comment]
> How is this summary different than the summary that shows up on the PR page? Duplicate?

Yeah, it's fully redundant. The PR comment is a better mechanism, but unfortunately we can't always create comments on PRs (like if the PR is coming in from a fork; see https://github.com/dart-lang/ecosystem/issues/56#issuecomment-1421581681). So this PR will create comments when it can, but always create workflow summaries, so at least the info's somewhere.

**Comment by devoncarew** [comment]
Here's what the summary on the workflow page looks like:

Screenshot 2023-02-09 at 11 15 25 AM

**Review by kevmoo — APPROVED** [review]

**Pull request merged**